src/screens/TypeScreen.test.tsx

```diff
@@ -55,8 +53,68 @@ describe('TypeScreen', () => {
     expect(addStub).toHaveBeenCalledWith({
       account: 'My Account',
       issuer: 'Issuer A',
-      secret: 'mysecret',
+      secret: 'MYSECRET',
       uid: '11-22-33-44',
     })
   })
+
+  describe('when malformed secret has been input', () => {
+    describe('when it contains un-authorised digits', () => {
+      beforeEach(() => {
+        const { getByLabelText } = setup()
+
+        fireEvent.changeText(getByLabelText('Issuer'), 'Issuer A')
+        fireEvent.changeText(getByLabelText('Secret'), 'mysecret!1')
+        fireEvent.changeText(getByLabelText('Account'), 'My Account')
+      })
+
+      it('should show an error message', () => {
+        const { getByText } = screen
+
+        expect(getByText('Invalid secret')).toBeDefined()
```

Comment:
if the getBy fails, it will throw an exception so here the text is always defined.
I suggest to change the `getBy` with a `queryBy` or the check the `toBeInTheDocument` instead of `toBeDefined`

Reply:
Well, the meaning of this test is exactly that: if the text is not there, it should throw an exception and fail.
`queryBy` usually will return a false positive test in case the text is not visible at all, because it will not throw an exception but will return `null` which is interpreted as `defined` for some reason.

Regarding the `toBeInTheDocument` we can't use it here because is part of `@testing-library/jest-dom` which (imho) shouldn't be used in native apps.

If is ok I would stick to this implementation, but open to change my mind :)

Reply:
For the queryBy, the check is just `receivedValue !== undefined`, so null is accepted. That's why you should change the check to `.not.toBeNull()`
BTW, this implementation is completely fine. My comment was just to give a better output if the test fails, instead of an exception 😄

Reply:
Make sense, to be fair 😄 
Will take it under consideration next time